pkg/gatherers/conditional/conditional_gatherer_test.go

```diff
@@ -327,6 +330,20 @@ func TestGetGatheringFunctions(t *testing.T) {
 				Err:             fmt.Errorf("invalid character 'n' looking for beginning of object key string"),
 			},
 		},
+		{
+			name:                 "remote configuration is available, can be parsed but does not pass validation",
+			endpoint:             "/gathering_rules",
+			remoteMockClient:     &MockGatheringRulesServiceClient{value: testRemoteConfigInvalid},
+			releaseVersionEnvVar: "1.2.3",
+			remoteConfigStatus: gatherers.RemoteConfigStatus{
+				ConfigAvailable: true,
+				ConfigValid:     false,
+				AvailableReason: AsExpectedReason,
+				ValidReason:     InvalidReason,
+				ConfigData:      []byte(testRemoteConfigInvalid),
```

Comment:
Would you recommend to store the default configuration in the archive instead?

Reply:
Not sure what is the best. I think as long as it is documented for anyone willing to use it should be fine. Just want to avoid misunderstandings:
- config valid -> use remote config -> write remote config
- config unparsable -> use default config -> write default config
- config not matching spec -> use default config -> write remote config

I think depends on what is the criteria we want to use. If we want to write whatever we really use, then we should write the default. If the criteria is to write whatever we receive, then if is not parsable maybe we should leave it empty.

Reply:
Updated in https://github.com/openshift/insights-operator/pull/1027